clerk/tokens_options.go

```diff
@@ -1,16 +1,43 @@
 package clerk
 
+import (
+	"strings"
+
+	sjose "go.step.sm/crypto/jose"
+)
+
 // VerifyTokenOption describes a functional parameter for the VerifyToken method
-type VerifyTokenOption func(*verifyTokenOptions)
+type VerifyTokenOption func(*verifyTokenOptions) error
 
 // WithAuthorizedParty allows to set the authorized parties to check against the azp claim of the session token
 func WithAuthorizedParty(parties ...string) VerifyTokenOption {
-	return func(o *verifyTokenOptions) {
+	return func(o *verifyTokenOptions) error {
 		authorizedParties := make(map[string]struct{})
 		for _, party := range parties {
 			authorizedParties[party] = struct{}{}
 		}
 
 		o.authorizedParties = authorizedParties
+		return nil
+	}
+}
+
+// WithJWTVerificationKey allows to set the JWK to use for verifying tokens without the need to download or cache any JWKs at runtime
+func WithJWTVerificationKey(key string) VerifyTokenOption {
+	return func(o *verifyTokenOptions) error {
+		// From the Clerk docs: "Note that the JWT Verification key is not in
+		// PEM format, the header and footer are missing, in order to be shorter
+		// and single-line for easier setup."
+		if !strings.HasPrefix(key, "-----BEGIN") {
+			key = "-----BEGIN PUBLIC KEY-----\n" + key + "\n-----END PUBLIC KEY-----"
+		}
+
+		jwk, err := sjose.ParseKey([]byte(key))
```

Comment:
I believe this won't work, you can parse a PEM formatted key to a JWK with the below code
```golang
func pemToJWK(key string) (*jose.JSONWebKey, error) {
	block, _ := pem.Decode([]byte(key))
	if block == nil {
		return nil, fmt.Errorf("invalid PEM encoded block")
	}

	if block.Type != "PUBLIC KEY" {
		return nil, fmt.Errorf("invalid key type, expected a Public Key")
	}

	rsaPublicKey, err := x509.ParsePKCS1PublicKey(block.Bytes)
	if err != nil {
		return nil, fmt.Errorf("failed to parse key: %w", err)
	}

	return &jose.JSONWebKey{Key: rsaPublicKey}, nil
}
```

Reply:
And https://github.com/clerkinc/clerk-sdk-go/pull/48/commits/e1685c3b302e159698bfd1015b863d90ef8c7572